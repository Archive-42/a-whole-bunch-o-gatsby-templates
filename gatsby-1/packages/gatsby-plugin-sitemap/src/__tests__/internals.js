const {
  withoutTrailingSlash,
  prefixPath,
  resolveSiteUrl,
  resolvePagePath,
  resolvePages,
  defaultFilterPages,
  serialize,
  pageFilter,
} = require(`../internals`)

const minimatch = require(`minimatch`)

beforeEach(() => {
  global.__PATH_PREFIX__ = ``
})

const SiteUrl = `https://example.net`
const TestPath = `/test/path/`

describe(`gatsby-plugin-sitemap internals tests`, () => {
  it(`withoutTrailingSlash should correct paths without final slash`, () => {
    const result = withoutTrailingSlash(`/`)
    expect(result).toEqual(`/`)

    const result2 = withoutTrailingSlash(TestPath)
    expect(result2).toStrictEqual(`/test/path`)
  })

  it(`prefixPath should correctly concatonate path`, () => {
    const result = prefixPath({
      url: TestPath,
      siteUrl: SiteUrl,
      pathPrefix: `/root`,
    })

    expect(result).toStrictEqual(`https://example.net/root/test/path/`)
  })

  it(`resolveSiteUrl should return SiteUrl`, () => {
    const result = resolveSiteUrl({
      site: { siteMetadata: { siteUrl: SiteUrl } },
    })
    expect(result).toStrictEqual(SiteUrl)
  })

  it(`resolvePagePath should return TestPath`, () => {
    const result = resolvePagePath({ path: TestPath })
    expect(result).toStrictEqual(TestPath)
  })

  it(`resolvePages should return sample pages`, () => {
    const pages = [{ path: `/page-1/` }, { path: `/page-2/` }]
    const result = resolvePages({ allSitePage: { nodes: pages } })
    expect(result).toEqual(pages)
  })

  it(`defaultFilterPages correctly detects path`, () => {
    const page = { path: TestPath }
    const result = defaultFilterPages(page, `/another/path/`, {
      resolvePagePath,
      minimatch,
      withoutTrailingSlash,
    })
    expect(result).toStrictEqual(false)

    const result2 = defaultFilterPages(page, TestPath, {
      resolvePagePath,
      minimatch,
      withoutTrailingSlash,
    })
    expect(result2).toStrictEqual(true)
  })

  it(`serialize should create expected object`, () => {
    const page = { path: TestPath }
    const result = serialize(page, { resolvePagePath })

    expect(result).toEqual({
      url: TestPath,
      changefreq: `daily`,
      priority: 0.7,
    })
  })

  it(`pageFilter should filter correctly`, () => {
    const { filteredPages: results } = pageFilter({
      allPages: [
        { path: `/404.html` },
        { path: `/to/be/removed` },
        { path: `/to/keep/1` },
        { path: `/to/keep/2` },
      ],
      filterPages: defaultFilterPages,
      excludes: [`/to/be/removed/`],
    })

    expect(results).toMatchSnapshot()
  })
})
