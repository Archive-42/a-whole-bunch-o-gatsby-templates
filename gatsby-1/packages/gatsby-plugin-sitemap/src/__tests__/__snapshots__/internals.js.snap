// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-sitemap internals tests pageFilter should filter correctly 1`] = `
Array [
  Object {
    "path": "/to/keep/1",
  },
  Object {
    "path": "/to/keep/2",
  },
]
`;
